vip-integrations.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+/**
+ * Plugin Name: VIP Integrations
+ * Description: Plugin loading for VIP plugin integrations.
+ * Author: Automattic
+ * License: GPL version 2 or later - http://www.gnu.org/licenses/old-licenses/gpl-2.0.html
+ */
+
+namespace Automattic\VIP\Integrations;
+
+if ( ! ( defined( 'VIP_BLOCK_DATA_API_ENABLE' ) && true === constant( 'VIP_BLOCK_DATA_API_ENABLE' ) ) ) {
+	// Do not automatically activate a plugin unless specified.
+	return;
+}
+
+const VIP_BLOCK_API_LATEST_VERSION = '1.0.0';
+
+class PluginLoadStrategy {
+	const PREFER_MU_PLUGINS_VERSION    = 'PREFER_MU_PLUGINS_VERSION';
+	const PREFER_CUSTOMER_CODE_VERSION = 'PREFER_CUSTOMER_CODE_VERSION';
+}
+
+$plugin_load_strategy = PluginLoadStrategy::PREFER_CUSTOMER_CODE_VERSION;
```

Comment:
We aren't changing this variable now .... so do we need any changes in `PluginLoadStrategy`?

Reply:
I wrote a little about this in the issue description:

> The `PluginLoadStrategy` used is largely to ensure that mu-plugins has the ability to decide load order if a plugin is both installed manually and enabled via constant. This is set to `PREFER_CUSTOMER_CODE_VERSION` currently which means if the Block Data API plugin is manually installed, that version will override the constant-enabled version.

Said another way, I wanted to make sure the the design of our plugin would allow it to be loaded before or after client code, depending on how we want VIP integration dependencies to work in the future. This was a design goal in including this plugin, and I wanted to make sure it wasn't set up in a way that made choosing later impossible.

Keeping in this code allowed me to test that this was working properly, and ensure the design worked. You're correct that it's hardcoded right now and doesn't really add any value as-is, but it's useful for ensuring the loading of the plugin is flexible.

I'm open to removing it, however, if you think it's no longer useful in this code.

Reply:
Removing seems better because it will make lines from 32 to 37 simpler ..... and we can add this preference as a comment on line 27

Reply:
Okay! That makes sense to me, and the use of the `VIP_BLOCK_DATA_API_LOADED` constant makes it pretty clear how we're avoiding double-loads. I removed the hardcoded value and all of the surrounding unused logic in https://github.com/Automattic/vip-go-mu-plugins/pull/4406/commits/ced2eade22da937867d756fdaf5ed1a53ce3ac66 / https://github.com/Automattic/vip-go-mu-plugins/pull/4406/commits/9fc3e89a2c1a102a133e501586503ad9175b3926 (comment).

Reply:
It's now a ton simpler.

Reply:
Great. BTW now I think it is easier to make this generic but completely your call 🙂